lib/logstash/outputs/s3.rb

```diff
@@ -147,6 +152,7 @@ def aws_s3_config
 
   def full_options
     aws_options_hash.merge(signature_options)
+    aws_options_hash.merge(endpoint_options)
```

Comment:
`.merge` returns a new hash rather than modifying `aws_options_hash` in-place. I'm not a Rubyist so I can't say what the prettiest way to fix this is but `aws_options_hash.merge(signature_options).merge(endpoint_options)` should do the trick.